sdk/src/main/java/io/radar/sdk/util/RadarSimpleLogBuffer.kt

```diff
@@ -0,0 +1,57 @@
+package io.radar.sdk.util
+
+import io.radar.sdk.Radar
+import io.radar.sdk.model.RadarLog
+import java.util.concurrent.LinkedBlockingDeque
+
+/**
+ * Basic Log Buffer implementation that is backed by a field.
+ */
+internal class RadarSimpleLogBuffer : RadarLogBuffer {
+
+    /**
+     * Concurrency-safe list of logs with a maximum capacity of 1000
+     */
+    private val list = LinkedBlockingDeque<RadarLog>(1000)
+
+    override fun write(level: Radar.RadarLogLevel, message: String) {
+        if (!list.offer(RadarLog(level, message))) {
+            purgeOldestLogs()
+            list.put(RadarLog(level, message))
+        }
+    }
+
+    override fun getFlushableLogsStash(): Flushable<RadarLog> {
+        val logs = mutableListOf<RadarLog>()
+        list.drainTo(logs)
+        return object : Flushable<RadarLog> {
+
+            override fun get(): List<RadarLog> {
+                return logs
+            }
+
+            override fun onFlush(success: Boolean) {
+                if (!success) {
+                    //reverse order to ensure the logs will purge correctly (oldest logs purged first)
+                    logs.reverse()
```

Comment:
`logs` is of type `LinkedBlockingDeque`, so I'm wondering what the `reverse` is needed for as elements can be added to / removed from each side?

Reply:
@jsani-radar `logs` is a locally-scoped variable (within this function) of type `List<RadarLog>`. `list` is the concurrency-safe class variable where these logs are being inserted in order. `RadarLog` has a `sort` method that is used to ensure the expected order. As @tjulien points out below, with the removal of the return statement, then the logs will continue to get added. This way, oldest logs are purged, but newest logs are kept, in order for future purging.